File: StringGen/java/stringgen/src/main/java/com/stringgen/application/StringGenerationService.java
package com.stringgen.application;

import com.stringgen.domain.CharacterSets;
import com.stringgen.domain.IStringGenerator;
import com.stringgen.domain.StringGenerator;
import com.stringgen.domain.Validations;

//Handles the use case of generating a string and storing it in the database.
public class StringGenerationService {
    
    private final IStringGenerator stringGenerator;

    public StringGenerationService() {
        stringGenerator =  new StringGenerator();
        
    }

    // Generates and saves a string based on the chosen option and length.
    // Validates length before generating, and stores the result in the repository.
    public String generateAndSaveString(int option, int length) {
        if (!Validations.isValidLength(length)) {
            throw new IllegalArgumentException("Invalid length.");
        }

        String characterSet = getCharacterSet(option);
        String generated = stringGenerator.generateString(characterSet, length);

        return generated;
    }

    // Returns the character set based on the user's choice.
    private String getCharacterSet(int option) {
        switch (option) {
            case 1 -> {
                return CharacterSets.UPPERCASE;
            }
            case 2 -> {
                return CharacterSets.LOWERCASE;
            }
            case 3 -> {
                return CharacterSets.DIGITS;
            }
            case 4 -> {
                return CharacterSets.UPPERCASE + CharacterSets.LOWERCASE + CharacterSets.DIGITS;
            }
            case 5 -> {
                return CharacterSets.UPPERCASE + CharacterSets.LOWERCASE + CharacterSets.DIGITS + CharacterSets.SIMBOLS;
            }
            default -> throw new IllegalArgumentException("Invalid option.");
        }
    }
}

File: Application/StringGenerationService.cs
using System;
using StringGen.Domain;

namespace StringGen.Application
{
    //Orchestrates the use case of generating a string.
    public class StringGenerationService
    {
        private readonly StringGenerator _stringGenerator;

        public StringGenerationService()
        {
            _stringGenerator = new StringGenerator();
        }

        //Generates a string based on the selected option and length and validates the length.
        public string GenerateAndSaveString(int option, int length)
        {
            if (!Validations.IsValidLength(length))
            {
                throw new ArgumentException("Invalid size");
            }

            string characterSet = GetCharacterSet(option);
            string generated = _stringGenerator.GenerateString(characterSet, length);

            return generated;
        }

        // Returns the character set based on the user's chosen option.
        private string GetCharacterSet(int option)
        {
            switch (option)
            {
                case 1:
                    return CharacterSets.Uppercase;
                case 2:
                    return CharacterSets.Lowercase;
                case 3:
                    return CharacterSets.Digits;
                case 4:
                    return CharacterSets.Uppercase + CharacterSets.Lowercase + CharacterSets.Digits;
                case 5:
                    return CharacterSets.Uppercase + CharacterSets.Lowercase + CharacterSets.Digits + CharacterSets.Simbols;
                default:
                    throw new ArgumentException("Invalid option.");
            }
        }
    }
}
